src/bundle/Controller/ContentController.php

```diff
@@ -381,4 +383,72 @@ public function updateMainTranslationAction(Request $request): Response
 
         return $this->redirectToRoute('ezplatform.dashboard');
     }
+
+    /**
+     * @param Symfony\Component\HttpFoundation\Request $request
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function updateVisibilityAction(Request $request): Response
+    {
+        $form = $this->createForm(ContentVisibilityUpdateType::class);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $this->submitHandler->handle($form, function (ContentVisibilityUpdateData $data) {
+                $contentInfo = $data->getContentInfo();
+                $newValue = $data->getVisible();
+
+                if ($contentInfo->isHidden && $newValue === false) {
```

Comment:
Could those two ifs be handled inside DataForm validation?

Reply:
then, the behaviour will change, I do not want to show an error to the user if everything is ok.